app/Observers/DeviceObserver.php

```diff
@@ -59,10 +59,15 @@ public function updated(Device $device): void
     public function deleted(Device $device): void
     {
         // delete rrd files
-        $host_dir = addcslashes(escapeshellarg(\Rrd::dirFromHost($device->hostname)), '\'');
-        $result = trim(shell_exec("bash -c '( [ ! -d $host_dir ] || rm -vrf $host_dir 2>&1 ) && echo -n OK'"));
-        if (! Str::endsWith($result, 'OK')) {
-            Log::debug("Could not $device->hostname files: $result");
+        $host_dir = \Rrd::dirFromHost($device->hostname);
+        try {
+            $result = File::deleteDirectory($host_dir);
+        } catch (\Exception $e) {
+            Log::error("Could not delete RRD files for: $device->hostname", [$e]);
+        }
+
+        if (isset($result) && $result == false) {
```

Comment:
Put this check inside the try block and you don't need to check isset ;)

Reply:
:upside_down_face: